tests/standalone_tests/test_cleanup_after_algorithm_execution.py

```diff
@@ -322,7 +322,10 @@ async def test_cleanup_after_uninterrupted_algorithm_execution(
     )
 
     # Releasing contextid, signals the Cleaner to start cleaning the contextid from the nodes
-    controller._cleaner.release_context_id(context_id=context_id)
+    controller._cleaner.cleanup_context_id(
+        context_id=context_id,
+        node_ids=[node_info.id for node_info in node_lanadscape_aggregator.get_nodes()],
+    )
```

Comment:
the purpose of the tests in this module is to test the asynchronous cleaning mechanism implemented by the Cleaner class. By replacing the call to the (_removed_) `release_context_id(..)` method with a call to the (_new_) `cleanup_context_id(..)` method,  the consequent call to the cleanup tasks becomes synchronous so it beats the purpose of all these integration tests. Right now all these tests only test that at some point, either after the successful completion of an algorithm execution, or a controller restart, or a node down incident, calling the cleanup task on the nodes, drops the relevant db tables __synchronously__. Which is something that could be tested with much simpler code and does not need to be tested under all these circumstances.

Reply:
All tests except the `test_cleanup_after_uninterrupted_algorithm_execution` are still testing the asynchronous cleaning mechanism implemented by the Cleaner class.
In every case that the node fails to cleanup the context id, an entry will be created and the asynchronous cleaning mechanism will be responsible for the cleanup of the specific context id.
The only test that was changed was the `test_cleanup_after_uninterrupted_algorithm_execution`.
On this specific test it is no longer needed to wait for the asynchronous cleaning mechanism so i removed the await part from the test.